mruby/src/value/mod.rs

```diff
@@ -75,6 +80,32 @@ impl Value {
     }
 }
 
+impl Clone for Value {
+    fn clone(&self) -> Self {
+        let ruby_type = self.ruby_type();
+        if ruby_type == Data {
+            unsafe {
+                let ptr = mem::transmute::<*mut c_void, Rc<RefCell<_>>>(self.value.value.p);
```

Comment:
@lopopolo i couldn't figure out what to use here? any "pointers" :trollface: 
also I have no idea this is correct or not. probably not

Reply:
this is the right idea! I think it is pretty dangerous to be mucking around with the value union directly though 😱 

There is an mruby macro called `DATA_PTR` which does the extraction for us. 

https://github.com/lopopolo/ferrocarril/blob/37034d455d75d39eae4be5ffdd458de98713e42b/mruby-sys/vendor/mruby-c078758/include/mruby/data.h#L49-L50

Because it is a macro, the generated bindings don't have a reference to it. Can you add a wrapper to the mruby-sys C extension? Something like this:

https://github.com/lopopolo/ferrocarril/blob/37034d455d75d39eae4be5ffdd458de98713e42b/mruby-sys/mruby-sys/include/mruby-sys/ext.h#L56

https://github.com/lopopolo/ferrocarril/blob/37034d455d75d39eae4be5ffdd458de98713e42b/mruby-sys/mruby-sys/src/mruby-sys/ext.c#L49-L51

Reply:
i also don't think we need the transmute here. We should be able to keep the ptr as a `*mut c_void`.

Reply:
@lopopolo can we use something like this instead? https://github.com/lopopolo/ferrocarril/blob/6e37851fe6421e6bd1a0fdc6b594d48bf76f1171/mruby-sys/src/ffi.rs#L2365-L2372
It seems to be doing exactly that

Reply:
never mind, that takes too many parameters 😅